### ISSUE TITLE
Mautic 5.2.1 API not working

### ISSUE THREAD

**Issue by alexhammerschmied** [mautic/mautic#14403]
### Mautic Series

5.2.x series

### Mautic installed version

5.2.1

### Way of installing

I downloaded a release from https://www.mautic.org/mautic-releases

### PHP version

8.2.26

### What browsers are you seeing the problem on?

Not relevant

### What happened?

Since the Update from Mautic 5.2.0 to 5.2.1 every API Call is returning a 500 error

### How can we reproduce this issue?

Step 1: Update to Mautic 5.2.1
Step 2: Create API OAuth2 Credentials
Step 3: Make an API Call to an endpoint ie /api/contacts

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by DogByteMarketing** [comment]
We've updated a few clients already and they have no issues with the API. Our lead capture software adds leads via a POST to the /contacts/new endpoint and contacts are flowing in as normal. If you're still having issues, could you answer a few questions?

- Have you tried any other endpoints such as a POST request to /contacts/new?
- You mentioned this happened during an upgrade, but also said you created new OAuth2 credentials, so to confirm did this issue also happen with the existing OAuth2 credentials you had, which got you to try new ones or was this the first time you setup OAuth2?
- Does the version in the footer show 5.2.1?
- Were there any errors during the upgrade?
- Was the upgrade done via composer?
- Did you clear cache after the upgrade?

**Comment by VisionovaDev** [comment]
I have the same problem. After the updare to 5.2.1 from the "ses+api" DNS Schema is not longer recognized. Sendin a test mail I got the error : "Unknown DSN scheme. Please make sure the mailer DSN is configured properly."
Please help

**Comment by DogByteMarketing** [comment]
> I have the same problem. After the updare to 5.2.1 from the "ses+api" DNS Schema is not longer recognized. Sendin a test mail I got the error : "Unknown DSN scheme. Please make sure the mailer DSN is configured properly." Please help

This is related to transports and not the API. We have an idea of the problem, but since this issue is unrelated to the current one, please open a new ticket if this doesn't work. This may be more suitable for the Mautic support forums though, since this seems more likely to be a configuration issue. As always, be sure to take a backup.

Are you updating with composer? If you replaced your existing composer file instead of updating it, you more than likely are missing the amazon-mailer dependency. You'll need to add this dependency back to your composer file and run
`composer update --with-dependencies`

If you don't have your original composer file you should be able to just run
`composer require symfony/amazon-mailer`

Then clear your cache.

**Comment by VisionovaDev** [comment]
I used "pabloveintimilla/mautic-amazon-ses".
So "composer require pabloveintimilla/mautic-amazon-ses"  solves the problem. Thanks

**Comment by mautibot** [comment]
This issue has been mentioned on **Mautic Forums**. There might be relevant details there:

https://forum.mautic.org/t/mautic-5-2-1-api-calls-returning-error-500-when-data-is-present/34572/2

**Comment by mautibot** [comment]
This issue has been mentioned on **Mautic Forums**. There might be relevant details there:

https://forum.mautic.org/t/mautic-5-2-1-api-calls-returning-error-500-when-data-is-present/34572/3

**Comment by alexhammerschmied** [comment]
Hi, this issue is NOT related to an email transport plugin. 
It is solely about the API itself ;)

**Comment by alexhammerschmied** [comment]
> We've updated a few clients already and they have no issues with the API. Our lead capture software adds leads via a POST to the /contacts/new endpoint and contacts are flowing in as normal. If you're still having issues, could you answer a few questions?
> 
> * Have you tried any other endpoints such as a POST request to /contacts/new?
> * You mentioned this happened during an upgrade, but also said you created new OAuth2 credentials, so to confirm did this issue also happen with the existing OAuth2 credentials you had, which got you to try new ones or was this the first time you setup OAuth2?
> * Does the version in the footer show 5.2.1?
> * Were there any errors during the upgrade?
> * Was the upgrade done via composer?
> * Did you clear cache after the upgrade?

yes i have tried a lot. In 5.2.0 everything worked as expected, now it doesn't anymore.

**Comment by mattkeenan** [comment]
you might want to check your database migrations; i had a failed migration when updating to 5.2.1. at first it looked like the upgrade went fine but then parts of Mautic stopped working (like adding new triggers which failed with a 500). Upon closer inspection the database migrations failed for 5.2.1. In particular with the _point_groups table.

**Comment by alexhammerschmied** [comment]
Hi all, after further investigation i found that some files in the ApiBundle were not up to date. It seems that the update process somehow wasn't updating all the files at some point. I simply replaced the ApiBundle with the new files from the repo and everything works as expected. Thank you all for your input :)
